api/clients/v2/examples/client_construction.go

```diff
@@ -18,28 +21,33 @@ import (
 	"github.com/Layr-Labs/eigenda/common/geth"
 	auth "github.com/Layr-Labs/eigenda/core/auth/v2"
 	"github.com/Layr-Labs/eigenda/core/eth"
+	"github.com/Layr-Labs/eigenda/core/eth/directory"
+	"github.com/Layr-Labs/eigenda/core/payments"
+	"github.com/Layr-Labs/eigenda/core/payments/clientledger"
+	"github.com/Layr-Labs/eigenda/core/payments/ondemand"
+	"github.com/Layr-Labs/eigenda/core/payments/reservation"
+	"github.com/Layr-Labs/eigenda/core/payments/vault"
 	"github.com/Layr-Labs/eigenda/encoding"
 	"github.com/Layr-Labs/eigenda/encoding/kzg"
 	"github.com/Layr-Labs/eigenda/encoding/kzg/prover/v2"
 	"github.com/Layr-Labs/eigenda/encoding/kzg/verifier/v2"
 	"github.com/Layr-Labs/eigensdk-go/logging"
 	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
 )
 
-// These constants are specific to the EigenDA holesky testnet. To execute the provided examples on a different
+// These constants are specific to the EigenDA Sepolia testnet. To execute the provided examples on a different
 // network, you will need to set these constants to the correct values, based on the chosen network.
 const (
-	ethRPCURL                  = "https://ethereum-holesky-rpc.publicnode.com"
-	disperserHostname          = "disperser-testnet-holesky.eigenda.xyz"
-	certVerifierRouterAddress  = "0x7F40A8e1B62aa1c8Afed23f6E8bAe0D340A4BC4e"
-	registryCoordinatorAddress = "0x53012C69A189cfA2D9d29eb6F19B32e0A2EA3490"
-	// These two addresses are no longer required for the Eth Client, but parameter is still being taken until we deprecate the flags
-	eigenDAServiceManagerAddress = ""
-	// operatorStateRetrieverAddress is still used for CertBuilder
-	operatorStateRetrieverAddress = "0x003497Dd77E5B73C40e8aCbB562C8bb0410320E7"
+	ethRPCURL         = "https://ethereum-sepolia-rpc.publicnode.com"
+	disperserHostname = "disperser-testnet-sepolia.eigenda.xyz"
+	// Contract registry address for Sepolia - this allows fetching all other contract addresses
+	contractRegistryAddress = "0x9620dC4B3564198554e4D2b06dEFB7A369D90257"
+	// CertVerifierRouter is not available from the contract registry and must be specified directly
+	certVerifierRouterAddress = "0x58D2B844a894f00b7E6F9F492b9F43aD54Cd4429"
```

Comment:
don't think this is true? pretty sure we have the default router in the directory?